feature-governance-impl/src/main/java/io/novafoundation/nova/feature_governance_impl/presentation/tindergov/cards/TinderGovCardsViewModel.kt

```diff
@@ -364,12 +291,38 @@ class TinderGovCardsViewModel(
 
     private fun observeReferendaAndAddToCards() {
         combine(interactor.observeReferendaAvailableToVote(this), basketFlow) { referenda, basket ->
-            ReferendaWithBasket(referenda, basket)
+            val referendaIds = referenda.map { it.id }
+            val summaries = referendaSummaryInteractor.getReferendaSummaries(referendaIds, viewModelScope)
+            val amounts = referenda.associate { it.id to it.getAmountModel() }
+
+            val referendaCards = referenda.mapToCardWithDetails(summaries, amounts)
+                .filterNotNull()
+
+            ReferendaWithBasket(referendaCards, basket)
         }
             .addNewReferendaToCards()
             .launchIn(this)
     }
 
+    private fun List<ReferendumPreview>.mapToCardWithDetails(
+        summaries: Map<ReferendumId, String>,
+        amounts: Map<ReferendumId, AmountModel?>
+    ): List<CardWithDetails?> {
```

Comment:
Again, this abuses nullable values too much. This whole ViewModel is extremely hard to read because you cant quickly understand why certain values can be null and when they can be null

For example this method is called only once with later "filterNotNull" called afterwards. Why does it even return nullable values at the first place?